src/Components/Server/src/Builder/ComponentsEndpointBuilderExtensions.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Components.Server;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Extensions for <see cref="ComponentEndpointBuilder"/>.
+    /// </summary>
+    public static class ComponentEndpointbuilderExtensions
+    {
+        /// <summary>
+        /// Adds <typeparamref name="TComponent"/> to the list of components registered with this <see cref="ComponentHub"/> instance.
+        /// The DOM selector associated with the <typeparamref name="TComponent"/> will default to the component type name in lowercase.
+        /// </summary>
+        /// <typeparam name="TComponent">The component type.</typeparam>
+        /// <param name="builder">The <see cref="ComponentEndpointBuilder"/>.</param>
+        /// <returns>The <paramref name="builder"/>.</returns>
+        public static ComponentEndpointBuilder AddComponent<TComponent>(this ComponentEndpointBuilder builder)
```

Comment:
My personal preference would be *not* to have this overload, and thus require people to specify the `selector` argument explicitly. Reasoning:

1. If people don't realise we are inferring a selector, they will get the wrong impression about how the DOM gets populated. They will probably think that `<app>` in their `Index.cshtml` is actually a usage of a Razor Component, since it is almost the same syntax.
2. Pretty soon we'll probably change `Index.cshtml` so that the `<app>` goes inside some kind of container, or maybe `<app>` remains as the container but we change the name of the root component from `App` to `Root` or something. This is going to be needed so that `<app>` remains in the output so we know where to inject the client-rendered content to replace the server-prerendered content.

@javiercn If you feel strongly about keeping the overload without `selector`, then so be it, but my preference would be to drop it.

Reply:
I don't, so I'll just remove them and make it explicit

Reply:
Got rid of them.